pkg/nsm/apis/nsmconnect/nsmconnect.proto

```diff
@@ -0,0 +1,120 @@
+syntax = "proto3";
+
+package nsmconnect;
+
+message Label {
+    map<string ,string> selector = 1;
+}
+
+message Metadata {
+    string name = 1;
+    string namespace = 2;
+    Label labels = 3;
+}
+
+message KernelInterface {
+   Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message VhostUserInterface{
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+message MemInterface {
+    Metadata metadata = 1;
+// Interface specific parameters go here.
+}
+
+enum InterfaceType {
+    kernel_interface = 0;
+    vhost_user = 1;
+    memif = 2;
+  }
```

Comment:
We need an SRIOV type here as well. The way that Ian Wells and I have been envisioning the SRIOV type is that we'll expose that into the pod with a name we generate and map to the exact node-local physical or virtual function of the NIC card. Make sense?

Reply:
@mestry Would we need to return different mechanism parameters for an SRIOV case?

Reply:
I will add it but please think about and let me know sriov specific parameters you would want to see in sriov interface specific struc.

Reply:
@edwarnicke We would definitely need to return different parameters. @sbezverk I will think about what those are and let you know.